Refine api 1

nzcodarnoc opened the pull request:
Make dockerfile do all the build so we can leverage it's cache.
Wire up rhubarb to an internal API
Expose rhubarb over express

mermop commented on README.md:
````diff
@@ -5,45 +5,53 @@ Exposing DanielSWolf / rhubarb-lip-sync as a service
 ```
 
-Visit http://localhost:3000/ and you should see the Rhubarb version installed.
+Visit <http://localhost:3000/> and you should see the Rhubarb version installed.
````
I did not know this was a shorthand for a self link in markdown. TIL!

mermop commented on src/api/get-shapes.ts:
```diff
@@ -0,0 +1,63 @@
+    return fileNames;
+  }
+  const cmd = `${rhubarbCmd} -r phonetic -f json -o ${fileNames.shapesFileName} ${fileNames.audioFileName}`;
+  console.info(`cmd: ${cmd}`);
```
Is this a standard way to log for a Node app?

nzcodarnoc replied:
Yep. It will be captured in any logs on the place it is running.

mermop commented on src/api/get-shapes.spec.ts:
```diff
@@ -0,0 +1,33 @@
+
+import getShapes from "./get-shapes";
+
+test("Passing a URL (of a WAV) returns the location of the mouth shape analysis and location of the downloaded WAV file.", async () => {
```
✨ Love to see tests

nzcodarnoc replied:
✨

mermop commented on src/api/get-shapes.spec.ts:
```diff
@@ -0,0 +1,33 @@
+
```
Is there a reason why there's a blank line here?

mermop commented on src/api/get-shapes.ts:
```diff
@@ -0,0 +1,63 @@
+  const parts: Array<string> = path.split("/");
+  const fileName: string = parts[parts.length - 1];
+  const things: Array<string> = fileName.split(".");
+  return things[0];
```
Is `things` a technical term here..

mermop replied:
`fileNameSections`?

nzcodarnoc replied:
Thank you! I've renamed it.

mermop commented on src/index.ts:
```diff
@@ -29,9 +27,25 @@ app.use(helmet());
+  res.send("OK. " + rhubarbVersion);
 });
 
+// TODO, tidy up api documentation
```
I'd recommend deleting these commented out things, if you need them again you can always check back in the commit history

nzcodarnoc replied:
Done

mermop commented on src/utils.ts:
```diff
@@ -0,0 +1,4 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+export const PORT = parseInt(String(process.env.PORT), 10);
```
Is it worth making this fail more explicitly if PORT isn't set, or set a sensible default?

nzcodarnoc replied:
I've added the validation library for env vars in nodejs

nzcodarnoc commented:
It's a quick and dirty way to log. The best practice is to use a library like "Winston" to better control where the output goes.

> On 3/04/2020, at 12:24 PM, Merrin Macleod <notifications@github.com> wrote:
> 
> ﻿
> @mermop commented on this pull request.
> 
> In src/api/get-shapes.ts:
> 
> > +  const audioFileName = await downloadAudio(request);
> +  const shapesFileName = tmp.tmpNameSync({
> +    prefix: "shapes_",
> +    postfix: ".txt"
> +  });
> +  const fileNames: AudioAndShapes = {
> +    audioFileName: String(audioFileName),
> +    shapesFileName: audioFileName ? shapesFileName : String(false),
> +    shapesID: shapesID(shapesFileName),
> +    valid: audioFileName ? true : false
> +  };
> +  if (!fileNames.valid) {
> +    return fileNames;
> +  }
> +  const cmd = `${rhubarbCmd} -r phonetic -f json -o ${fileNames.shapesFileName} ${fileNames.audioFileName}`;
> +  console.info(`cmd: ${cmd}`);
> Is this a standard way to log for a Node app?
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

nzcodarnoc commented:
Tests give me life (true story)

> On 3/04/2020, at 12:25 PM, Merrin Macleod <notifications@github.com> wrote:
> 
> ﻿
> @mermop commented on this pull request.
> 
> In src/api/get-shapes.spec.ts:
> 
> > @@ -0,0 +1,33 @@
> +
> +import getShapes from "./get-shapes";
> +
> +test("Passing a URL (of a WAV) returns the location of the mouth shape analysis and location of the downloaded WAV file.", async () => {
> ✨ Love to see tests
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

nzcodarnoc commented:
Nope

> On 3/04/2020, at 12:25 PM, Merrin Macleod <notifications@github.com> wrote:
> 
> ﻿
> @mermop commented on this pull request.
> 
> In src/api/get-shapes.spec.ts:
> 
> > @@ -0,0 +1,33 @@
> +
> Is there a reason why there's a blank line here?
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

nzcodarnoc commented:
I am so open to a better variable name here. Ideas?

> On 3/04/2020, at 12:26 PM, Merrin Macleod <notifications@github.com> wrote:
> 
> ﻿
> @mermop commented on this pull request.
> 
> In src/api/get-shapes.ts:
> 
> > +import tmp from "tmp";
> +import { rhubarbCmd } from "../utils";
> +import { exec } from "child_process";
> +
> +interface AudioAndShapes {
> +  shapesFileName: string;
> +  audioFileName: string;
> +  shapesID: string;
> +  valid: boolean;
> +}
> +
> +const shapesID = (path: string): string => {
> +  const parts: Array<string> = path.split("/");
> +  const fileName: string = parts[parts.length - 1];
> +  const things: Array<string> = fileName.split(".");
> +  return things[0];
> Is things a technical term here..
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

nzcodarnoc commented:
Sure thing 

> On 3/04/2020, at 12:27 PM, Merrin Macleod <notifications@github.com> wrote:
> 
> ﻿
> @mermop commented on this pull request.
> 
> In src/index.ts:
> 
> >  });
>  
> +// TODO, tidy up api documentation
> I'd recommend deleting these commented out things, if you need them again you can always check back in the commit history
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

nzcodarnoc commented:
For sure

> On 3/04/2020, at 12:29 PM, Merrin Macleod <notifications@github.com> wrote:
> 
> ﻿
> @mermop commented on this pull request.
> 
> In src/utils.ts:
> 
> > @@ -0,0 +1,4 @@
> +import * as dotenv from "dotenv";
> +dotenv.config();
> +export const PORT = parseInt(String(process.env.PORT), 10);
> Is it worth making this fail more explicitly if PORT isn't set, or set a sensible default?
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
